### PR TITLE
[corlib] Fix tests for watchOS.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

**Comment by vargaz** [comment]
LGTM

File: mcs/class/corlib/Test/System.Reflection/AssemblyTest.cs [hunk 1]
```diff
@@ -182,7 +182,10 @@ public void GetEntryAssembly ()
 			// note: only available in default appdomain
 			// http://weblogs.asp.net/asanto/archive/2003/09/08/26710.aspx
 			// Not sure we should emulate this behavior.
-#if !MONODROID
+#if __WATCHOS__
+			Assert.IsNull (Assembly.GetEntryAssembly (), "GetEntryAssembly");
+			Assert.IsTrue (AppDomain.CurrentDomain.IsDefaultAppDomain (), "!default appdomain");
+#elif !MONODROID
```
Comment (akoeplinger) [inline mcs/class/corlib/Test/System.Reflection/AssemblyTest.cs]: why not just `#if !MONODROID && !__WATCHOS__` as the #else is the same as this code here?

File: mcs/class/corlib/Test/System/EnvironmentTest.cs [hunk 2]
```diff
@@ -156,9 +156,13 @@ public void GetEnvironmentVariables_NewlySet ()
 		public void GetCommandLineArgs ()
 		{
 			string[] args = Environment.GetCommandLineArgs ();
+#if !__WATCHOS__
 			Assert.IsNotNull (args, "not null");
 			Assert.IsTrue (((args.Length > 0) && (args.Length < 256)), "reasonable");
 			Assert.IsNotNull (args [0], "application");
+#else
+			Assert.AreEqual (0, args.Length, "length");
+#endif
```
Comment (akoeplinger) [inline mcs/class/corlib/Test/System/EnvironmentTest.cs]: I think we should just edit GetCommandLineArgs to return an empty collection instead so consumers don't need to deal with these differences.

We even have code in the BCL that relies on this not being null: https://github.com/mono/mono/blob/91734968b83a0971c041254286cdc903b26564e2/mcs/class/corlib/System/Environment.cs#L129

**Comment by akoeplinger** [comment]
LGTM.

@monojenkins merge

**Comment by vargaz** [comment]
The failures are unrelated.

**Pull request merged**